src/perform_step/adams_bashforth_moulton_perform_step.jl

```diff
@@ -1536,3 +1536,59 @@ end
     cache.ϕstar_nm1[i] .= ϕstar_n[i]
   end
 end
+
+
+# ABCN2
+
+function initialize!(integrator,cache::ABCN2ConstantCache)
+  integrator.kshortsize = 2
+  integrator.k = typeof(integrator.k)(integrator.kshortsize)
+  integrator.fsalfirst = 
+  integrator.f.f1(integrator.uprev,integrator.p,integrator.t) +
+  integrator.f.f2(integrator.uprev,integrator.p,integrator.t) # Pre-start fsal
+
+  # Avoid undefined entries if k is an array of arrays
+  integrator.fsallast = zero(integrator.fsalfirst)
+  integrator.k[1] = integrator.fsalfirst
+  integrator.k[2] = integrator.fsallast
+end
+
+function perform_step!(integrator,cache::ABCN2ConstantCache,repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack k2,uf,κ,tol = cache
+  cnt = integrator.iter
+  f1 = integrator.f.f1
+  f2 = integrator.f.f2
+  if cnt == 1
+    halfdt = dt/2
+    k = f1(uprev + halfdt*integrator.fsalfirst, p, t+halfdt) + f2(uprev + halfdt*integrator.fsalfirst, p, t+halfdt)
```

Comment:
No reason to do midpoint here for the extra function call. Euler is fine, and there's other theoretical justifications for using Euler here.

Reply:
I thought it would be good to use 2nd order method for starting value as `ABCN2` is a 2nd order method.

Reply:
Euler is 1st order globally, but in a single step it's second order. Generally that's the result, so you can always use a starter of one order less (that's why RK4 for 5th order Adams is fine). If you assume that the previous derivative and the current derivative are the same, AB2 = forward Euler, so it's a very natural starting assumption, and you can show this is stable.

Reply:
Okay